### PR TITLE
Main menu save load

### PR DESCRIPTION
**Pull request by chern042**
Gave functionality to the "Save" button on pause. Also added a button on the main menu to allow you to load a saved game. This is a very basic version that saves player health, level, mana, and position. Also cleaned up the code a bit to remove some repeated code and completely remove "PlayerManager" which was useless.

### CODE REVIEW

**Comment by chern042** [comment]
Can leave the "savedGameIndicator" code as-is for now so i can complete that later.

**Comment by RossBaumgartner** [comment]
How can I test this? Not too sure how to load a save

File: Assets/Scripts/Controllers/PlayerController.cs
Comment (AndreiBaetica) [inline Assets/Scripts/Controllers/PlayerController.cs:31]: There's some duplicate lines of code in each clause of this if statement (ex. NPCMask = LayerMask.GetMask("NPC");). These should be pulled outside the if statement.

File: Assets/Scripts/Controllers/CharController.cs
Comment (AndreiBaetica) [inline Assets/Scripts/Controllers/CharController.cs:11]: why remove the default value?

File: Assets/Scripts/Controllers/CharController.cs
Comment (AndreiBaetica) [inline Assets/Scripts/Controllers/CharController.cs:200]: This code block should be reverted. It does the same thing as the one-liner it replaces

File: Assets/Scripts/Controllers/CharController.cs
Comment (AndreiBaetica) [inline Assets/Scripts/Controllers/CharController.cs:46]: I see this logic is being moved from the parent to the child classes, but I don't understand why

File: Assets/Scripts/Controllers/EnemyController.cs
Comment (AndreiBaetica) [inline Assets/Scripts/Controllers/EnemyController.cs:35]: this logic shoud already be covered by TakeDamage()

File: Assets/Scripts/Controllers/PlayerController.cs
Comment (AndreiBaetica) [inline Assets/Scripts/Controllers/PlayerController.cs:80]: this logic shoud already be covered by TakeDamage()

**Comment by AndreiBaetica** [comment]
I notice you're using the textmesh library. Do we need to do anything with licensing?

**Comment by RossBaumgartner** [comment]
> I notice you're using the textmesh library. Do we need to do anything with licensing?

I think I saw licences in the added files, and they said that it could be used even for profit so I think we are good.

We will probably replace most of the text anyway with a different font or art

**Comment by AndreiBaetica** [comment]
> > I notice you're using the textmesh library. Do we need to do anything with licensing?
> 
> I think I saw licences in the added files, and they said that it could be used even for profit so I think we are good.
> 
> We will probably replace most of the text anyway with a different font or art

Yeah, the main thing is the text shader. We'll have to do some tests once we have our own font to see if it even needs a shader

File: Assets/Scripts/Controllers/CharController.cs
Comment (chern042) [reply Assets/Scripts/Controllers/CharController.cs:11]: Because enemies and players will have different max healths set by their controllers.

File: Assets/Scripts/Controllers/CharController.cs
Comment (chern042) [reply Assets/Scripts/Controllers/CharController.cs:46]: Because currentHealth is only set to maxHealth on the player when there is a save file. The enemies always start with mnax health, obviously, so i had to break it apart.

File: Assets/Scripts/Controllers/CharController.cs
Comment (AndreiBaetica) [reply Assets/Scripts/Controllers/CharController.cs:11]: That's true, but it's still good practice to have a default to  fall back on

Comment (chern042) [reply Assets/Scripts/Controllers/CharController.cs:11]: Im going to move everything to use its getters/setters.

**Review by RossBaumgartner — APPROVED** [review]
tested and working as intended

it looks like your most recent changes also address Andreis comments, but I would wait for his review before merging

File: Assets/Scripts/Controllers/EnemyController.cs
Comment (SebastienGorman) [inline Assets/Scripts/Controllers/EnemyController.cs:1]: duplicate?

**Review by SebastienGorman — COMMENTED** [review]
Looks good overall, tested and works well. Only found very small duplicate to fix.

**Review by SebastienGorman — APPROVED** [review]

File: Assets/Scripts/Controllers/CharController.cs
Comment (AndreiBaetica) [reply Assets/Scripts/Controllers/CharController.cs:46]: Right, that makes sense

**Review by AndreiBaetica — APPROVED** [review]
reviewed and tested, nice work my dude

**Pull request merged**
